### PR TITLE
luci-app-apfree-wifidog: add luci-app-apfree-wifidog

### PR DESCRIPTION
**Pull request by liudf0716**
add luci-app-apfree-wifidog for apfree-wifidog
https://github.com/openwrt/packages/pull/21203

### CODE REVIEW

File: applications/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		o = s.option(form.Value, 'check_interval', _('Check Interval(s)'), _('The interval to check the status of the gateway'));
+		o.rmempty = false;
+		o.datatype = 'uinteger';
+		// wired_passed
+		o = s.option(form.Flag, 'wired_passed', _('Wired Passed'), _('Wired client will be passed without authentication'));
+		o.rmempty = false;
+
+		return m.render();
+	}
+});
```
Comment (feckert) [inline applications/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js:99]: Could your please replace the comments `// <option nam>` with an empty line. This is not needed. And could you also move the description into an new line so we do not hit the 80 char line boundary.

File: applications/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+				E('fieldset', { class: 'cbi-section'}, [
+					E('p', { id: 'service_status' },
+						_('Collecting data ...'))
+				])
+			);
+		}
+
+		s = m.section(form.TypedSection, "wifidogx", _("ApFree-WiFiDog"), _("ApFree-WiFiDog Settings"));
+		s.anonymous = true;
+		// add client settings
```
Comment (feckert) [inline applications/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js:66]: Please remove this line

File: applications/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+return view.extend({
+	render: function() {
+		var m, s, o;
+
+		m = new form.Map('wifidogx', _('ApFree-WiFiDog'));
+		m.description = _("apfree-wifiodg is a Stable & Secure captive portal solution.");
+
+		// add kcptun-client status section and option
```
Comment (feckert) [inline applications/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js:45]: Please remove this line

**Review by feckert — CHANGES_REQUESTED** [review]
Everything else looks good.

File: applications/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js (hunk 3 above)
Comment (liudf0716) [reply applications/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js]: @feckert fixed, thanks!

**Comment by liudf0716** [comment]
@feckert What else can I do to get my pull request approved?

**Review by systemcrash — CHANGES_REQUESTED** [review]

File: applications/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+		o.rmempty = false;
+		o.datatype = 'string';
+		
+		o = s.option(form.Value, 'check_interval', _('Check Interval(s)'), 
+			_('The interval to check the status of the gateway'));
+		o.rmempty = false;
+		o.datatype = 'uinteger';
+		
+		o = s.option(form.Flag, 'wired_passed', _('Wired Passed'), 
+			_('Wired client will be passed without authentication'));
```
[the remark this answers is not in the document]
Comment (liudf0716) [reply applications/luci-app-apfree-wifidog/htdocs/luci-static/resources/view/wifidogx.js:101]: @systemcrash yes.  a wired client does not require authentication to access the network.

**Review by systemcrash — APPROVED** [review]

**Comment by liudf0716** [comment]
@systemcrash @feckert 
What other actions should I take? I noticed the "work needed" label on the pull request. thanks

**Comment by systemcrash** [comment]
> @systemcrash @feckert What other actions should I take? I noticed the "work needed" label on the pull request. thanks

Change that string and we can merge

**Comment by liudf0716** [comment]
> > @systemcrash @feckert What other actions should I take? I noticed the "work needed" label on the pull request. thanks
> 
> Change that string and we can merge

I'm sorry, I'm not sure which specific string you're referring to. Could you please provide more context or identify the exact string you'd like me to change? I was under the impression that I had already made all the necessary changes, but I'm happy to double-check and make any further adjustments.

**Comment by systemcrash** [comment]
> I'm sorry, I'm not sure which specific string you're referring to. Could you please provide more context or identify the exact string you'd like me to change? I was under the impression that I had already made all the necessary changes, but I'm happy to double-check and make any further adjustments.

Look through this PR again, and resolve the outstanding review.

**Pull request closed without merging**